src/EnergyPlus/AirflowNetwork/src/Elements.cpp

```diff
@@ -3690,6 +3681,1015 @@ namespace AirflowNetwork {
+        // na
+
+        // Return value
+        Real64 psz;
+
+        // Locals
+        // SUBROUTINE ARGUMENT DEFINITIONS:
+
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        // na
+
+        // INTERFACE BLOCK SPECIFICATIONS
+        // na
+
+        // DERIVED TYPE DEFINITIONS
+        // na
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        Real64 dz;
+        Real64 rho;
+
+        dz = z - z0;
+        rho = (Rho0 + beta * dz / 2.0);
+        psz = -Pz0 * (1.0 - std::exp(-dz * rho * g / Pz0)); // Differential pressure from z to z0 [Pa]
+
+        return psz;
+    }
+
+    void DetailedOpeningSolver::lclimb(EnergyPlusData &state,
+                                       Real64 const G,   // gravity field strength [N/kg]
+                                       Real64 &Rho,      // Density link level (initialized with rho zone) [kg/m3]
+                                       Real64 const Z,   // Height of the link above the zone reference [m]
+                                       Real64 &T,        // temperature at link level [C]
+                                       Real64 &X,        // absolute humidity at link level [kg/kg]
+                                       Real64 &Dp,       // Stackpressure to the linklevel [Pa]
+                                       int const zone,   // Zone number
+                                       Real64 const PZ,  // Zone Pressure (reflevel) [Pa]
+                                       Real64 const Pbz, // Barometric pressure at entrance level [Pa]
+                                       Real64 &RhoDr     // Air density of dry air on the link level used
+    )
+    {
+
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Lixing Gu
+        //       DATE WRITTEN   Oct. 2005
+        //       MODIFIED       na
+        //       RE-ENGINEERED  This subroutine is revised based on subroutine IClimb from COMIS
+
+        // PURPOSE OF THIS SUBROUTINE:
+        // This function the differential pressure from the reflevel in a zone To Z, the level of a link
+
+        // METHODOLOGY EMPLOYED:
+        // na
+
+        // REFERENCES:
+        // na
+
+        // USE STATEMENTS:
+        // na
+
+        // Locals
+        // SUBROUTINE ARGUMENT DEFINITIONS:
+        // for the concentration routine [kg/m3]
+
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        // na
+
+        // INTERFACE BLOCK SPECIFICATIONS
+        // na
+
+        // DERIVED TYPE DEFINITIONS
+        // na
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        Real64 H;        // Start Height of the layer
+        Real64 BetaT;    // Temperature gradient of this layer
+        Real64 BetaXfct; // Humidity gradient factor of this layer
+        Real64 BetaCfct; // Concentration 1 gradient factor of this layer
+        Real64 X0;
+        Real64 P;
+        Real64 Htop;
+        Real64 Hbot;
+        Real64 Rho0;
+        Real64 Rho1;
+        Real64 BetaRho;
+        static int L(0);
```

Comment:
Are these really intended to be static, i.e., the initial value of L is the final value from the previous call? Should probably move that to `state` in that case, since it is basically shadow state.

Reply:
As far as I know, yes, it is intended to be static. And it will be dealt with.